cfg/builder/builder_walk.cc

```diff
@@ -929,6 +929,16 @@ BasicBlock *CFGBuilder::walk(CFGContext cctx, ast::ExpressionPtr &what, BasicBlo
                 ret = current;
             },
 
+            [&](ast::KeepForIDE &keep) {
+                // Like ast::Cast's expansion, this is the only way to ensure that the
+                // constant expression here makes it into the CFG for LSP to hit on
+                // its location.
+                LocalRef deadSym = cctx.newTemporary(core::Names::keepForIde());
+                current = walk(cctx.withTarget(deadSym), keep.expr, current);
+                current->exprs.emplace_back(deadSym, core::LocOffsets::none(), make_insn<KeepAlive>(deadSym));
```

Comment:
Why are we using `core::LocOffsets::none()` here if we have `keep->loc`?

Reply:
(I'm generally somewhat paranoid of rare usages of a loc that doesn't exist, because so much code in Sorbet secretly assumes that they exist.)

Reply:
Eh, we can use `keep->loc`, which is zero-length, but the assignment here really doesn't have a loc.